repo/repo

```diff
@@ -653,7 +653,11 @@ packageName="repo${packageName//[^[:alpha:][:digit:].-]/}"
 packageVersion=$(date +"%s")
 
 # prepare zip contents in temp directory
-tmpDir=`mktemp -d -t repo.XXX`
+if [[ $(uname) =~ CYGWIN_NT.* ]]; then
+    tmpDir=`mktemp -d -p .. -t repo.XXX`
```

Comment:
IIUC, this creates the temporary directory in the parent folder of the current working dir? I wonder if this is a safe approach?

Reply:
In practice, this creates a temporary directory under the root of your checkout tree, in my case C:\dev\git. Anything but relative path would not work for me. I had to resort to this approach because curl 7.55.1 from my Cygwin would fail to open files created under /tmp/repo.XXX, which is the current behavior. This seems like a recent change in curl's behavior, similar to restrictions on file access by web servers.

Reply:
Do you get the "Warning: Couldn't read data from file" error like in https://stackoverflow.com/questions/16615900/sending-json-files-in-curl-requests-with-absolute-or-relative-paths ?

Reply:
I get "couldn't open file"
uploading /apps/trainingproject/components/content/helloworld/helloworld.html to http://localhost:4502
FIND: Invalid switch
      895  07-22-2019 21:07   jcr_root/apps/trainingproject/components/content/helloworld/helloworld.html
curl: (26) couldn't open file "/tmp/repo.3Ar/pkg.zip"
repo: failed to upload package: